pyopencl/__init__.py

```diff
@@ -1540,6 +1548,10 @@ def choose_devices(interactive: Optional[bool] = None,
     :returns: a list of :class:`Device` instances.
     """
 
+    # This must be done before PoCL is actually loaded
+    from pyopencl.characterize import setup_pocl_cache_dirs
+    setup_pocl_cache_dirs()
```

Comment:
Could you explain why this is here? This seems to defeat inter-run caching, which, in my book, is definitely desirable.

Reply:
This is a workaround for the intermittent failures we have seen as part of #738. 

I couldn't think of a way to ensure PID-local caches that also support inter-run caching, without resorting to using MPI rank numbers (which wouldn't even help for the CI errors seen in #738).

Reply:
Before we go in this direction, I think we should engage with the pocl folks and least file an issue to see what they think about fixing concurrent access to their caches.

Reply:
Reported here: https://github.com/pocl/pocl/issues/1480, including a small reproducer.

Reply:
Reverted in [b368eba](https://github.com/inducer/pyopencl/pull/749/commits/b368eba29f942b3e1939489f131a3799bbfbe37e)